src/main/java/com/google/sps/servlets/SeeJourneyServlet.java

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.sps.data.UserData;
+import com.google.gson.Gson;
+import java.util.List;
+import java.io.IOException;
+import java.util.ArrayList;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.FirebaseAuthentication;
+import com.google.sps.data.FirebaseSingletonApp;
+import com.google.sps.data.GoogleMapsPathFinder;
+import com.google.sps.data.JourneyHandler;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that handles requests for the delivery journeys of users.
+ */
+@WebServlet("/see-journeys")
+public class SeeJourneyServlet extends HttpServlet {
+  private FirebaseAuthentication firebaseAuth;
+
+  @Override
+  public void init() throws ServletException {
+    // initialize firebase app
+    try {
+      setFirebaseAuth(new FirebaseAuthentication(FirebaseSingletonApp.getInstance()));
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
Would it make more sense here to rethrow the IOException here? I am not sure what will happen when this silently catches an IOException - presumably the app will just run as usual (since it does not know about the exception initialising this servlet), but Firebase wont work?

Reply:
Yes, the only problem is that` init()` can't throw ` IOException`. 
If the error is catched, I think you're right, the app will work until Firebase will be accessed, at which point it will throw an error. 
I can change the code to set the Firebase instance in doGet() instead.